admin-frontend/app_singleapp/lib/widgets/user/signin/signin_widget.dart

```diff
@@ -193,29 +204,28 @@ class _SetupPage1ThirdPartyProviders extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final external = bloc.identityProviders.has3rdParty;
-
-    final children = <Widget>[];
-
     if (external) {
-      bloc.identityProviders.externalProviders.forEach((provider) {
-        children.add(InkWell(
-          mouseCursor: SystemMouseCursors.click,
-          onTap: () {
-            selectedExternalProviderFunc(provider);
-          },
-          child: Image.asset(
-              bloc.identityProviders.externalProviderAssets[provider]!),
-        ));
-        children.add(Padding(
-          padding: const EdgeInsets.fromLTRB(0, 16, 0, 10),
-          child: Text('or login with a username and password',
-              style: Theme.of(context).textTheme.caption),
-        ));
-      });
+      return Column(
+        children: [
+          for (dynamic provider in bloc.identityProviders.externalProviders)
+            Container(
+              height: 48,
+              child: Padding(
+                padding: const EdgeInsets.only(top: 12.0),
+                child: SignInButton(
+                    provider == 'oauth2-google'
+                        ? Buttons.GoogleDark
+                        : provider == 'oauth2-github'
```

Comment:
this logic would be simplified with a Map<String, IdentityConfig> class where IdentityConfig includes the button and text used (IdentityConfig is a class I just made up which would have at least two fields - the button and the text shown).